### PR TITLE
Feature/add cloudprovider sap converged cloud

### PR DESCRIPTION
**Pull request by killermoehre**
SAP Converged Cloud is a international cloud provider inside SAP SE. It runs OpenStack on VMware. Their upstream is at https://github.com/sapcc

Due to bug https://bugs.launchpad.net/cloud-init/+bug/1815990 we're not able to force the use of the OpenStack datasource. So we need to become an official provider.

### CODE REVIEW

File: cloudinit/sources/DataSourceOpenStack.py [hunk 1]
```diff
@@ -29,7 +29,8 @@
 DMI_PRODUCT_COMPUTE = 'OpenStack Compute'
 VALID_DMI_PRODUCT_NAMES = [DMI_PRODUCT_NOVA, DMI_PRODUCT_COMPUTE]
 DMI_ASSET_TAG_OPENTELEKOM = 'OpenTelekomCloud'
-VALID_DMI_ASSET_TAGS = [DMI_ASSET_TAG_OPENTELEKOM]
+DMI_ASSET_TAG_SAPCCLOUD = 'SAP CCloud VM'
```
Comment (OddBloke) [inline cloudinit/sources/DataSourceOpenStack.py]: Is there any documentation we could link to for this value? (I totally understand if not, but would be good to have if possible.)

File: doc/rtd/topics/datasources/openstack.rst [hunk 2]
```diff
@@ -19,7 +19,7 @@ checks the following environment attributes as a potential OpenStack platform:
 
    * **/proc/1/environ**: Nova-lxd contains *product_name=OpenStack Nova*
    * **DMI product_name**: Either *Openstack Nova* or *OpenStack Compute*
-   * **DMI chassis_asset_tag** is *OpenTelekomCloud* or *OpenStack Nova*
+   * **DMI chassis_asset_tag** is *OpenTelekomCloud*, *SAP CCloud VM* or *OpenStack Nova*
```
Comment (OddBloke) [inline doc/rtd/topics/datasources/openstack.rst]: We generally keep the docs wrapped at 80 chars, could you wrap this line, please?

File: cloudinit/sources/DataSourceOpenStack.py (hunk 1 above)
Comment (killermoehre) [reply cloudinit/sources/DataSourceOpenStack.py]: The "documentation" is here https://github.com/sapcc/helm-charts/blob/master/openstack/nova/values.yaml#L242

Comment (OddBloke) [reply cloudinit/sources/DataSourceOpenStack.py]: Aha, I poked around in the nova project over there, didn't think to look in helm-charts. Even a pointer to that would be good (ideally using a permalink, so https://github.com/sapcc/helm-charts/blob/c72c8a40a68f718a372607c2b1984a6f91f74291/openstack/nova/values.yaml#L242).

**Pull request merged**